hazelcast/src/main/java/com/hazelcast/sql/impl/connector/LocalPartitionedMapConnector.java

```diff
@@ -85,6 +85,7 @@ public Table createTable(
         long estimatedRowCount = estimatePartitionedMapRowCount(nodeEngine, context, mapName);
         List<MapTableIndex> indexes = container != null ? getPartitionedMapIndexes(container, fields) : emptyList();
         int distributionFieldOrdinal = container != null ? getPartitionedMapDistributionField(container, context, fields) : -1;
+        boolean nativeMemoryEnabled = nodeEngine.getConfig().getNativeMemoryConfig().isEnabled();
```

Comment:
This flag looks suspicious to me. Even though an instance may have native memory enabled, this doesn't mean that the concrete map use NATIVE format.

Reply:
My impression was that once it is enabled, it is used for all maps. But, OK, if we still for some maps can use on-heap format, I'll initialize it on the map level.

However, here we don't know which format is used, because we might have `container = null`. I also see a comment "TODO: do we want to keep it? maps are auto discovered...".  So, the question here, do we have to pass nativeMemoryEnabled from the caller?

Reply:
If `container == null`, then the map cannot be used. We should not focus a lot on this specific part of the code too much, because `LocalPartitinedMapConnector` is research code. The following pseudocode should be sufficient:
```
nativeMemoryEnabled = container != null && nodeEngine.nativeMemoryEnabled() && containter.format == NATIVE
```